base/some.jl

```diff
@@ -145,3 +145,12 @@ macro something(args...)
     something = GlobalRef(Base, :something)
     return :($something($expr))
 end
+
+==(a::Some, b::Some)::Bool = a.value == b.value
+# seperated out to ensure the above assertion works
+==(::Some{Missing}, ::Some) = missing
+==(::Some, ::Some{Missing}) = missing
+==(::Some{Missing}, ::Some{Missing}) = missing
+
```

Comment:
This seems sufficient to me:
```suggestion
==(a::Some, b::Some) = a.value == b.value
```

(But I don't know whether everybody is OK with having `==` call `==` recursively rather than `isequal`.)

Reply:
That seems like the least surprising thing, imo.

Reply:
I originally had this simpler implementation, but got ambiguity errors from some other place. I think it was the `missings.jl` tests.